src/System.Private.CoreLib/shared/System/Type.cs

```diff
@@ -375,7 +376,7 @@ public static Binder DefaultBinder
                 if (s_defaultBinder == null)
                 {
                     DefaultBinder binder = new DefaultBinder();
-                    Interlocked.CompareExchange<Binder>(ref s_defaultBinder, binder, null);
+                    Interlocked.CompareExchange<Binder?>(ref s_defaultBinder!, binder, null);
```

Comment:
>! [](start = 76, length = 1)

Why is this needed?  Shouldn't `s_defaultBinder` instead be declared as `Binder?`?

Reply:
nullable for <Binder?> is for last argument `null`, here `s_defaultBinder` is not declared as nullable. Seems better to declare `s_defaultBinder` nullable and bang returning value

Reply:
It shouldn't complain if s_defaultBinder is `Binder?`

Reply:
`s_defaultBinder` is `Binder` type, i might didn't understood your question correctly